server/src/main/java/org/opensearch/extensions/rest/RestSendToExtensionAction.java

```diff
@@ -0,0 +1,136 @@
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.Collections.unmodifiableList;
+
+/**
+ * An action that forwards REST requests to an extension
+ */
+public class RestSendToExtensionAction extends BaseRestHandler {
+
+    private static final Logger logger = LogManager.getLogger(RestSendToExtensionAction.class);
+
+    private final List<Route> routes;
+    private final String uriPrefix;
+    private final DiscoveryExtension discoveryExtension;
+    private final TransportService transportService;
+
+    /**
+     * Instantiates this object using a {@link RegisterRestActionsRequest} to populate the routes.
+     *
+     * @param restActionsRequest A request encapsulating a list of Strings with the API methods and URIs.
+     * @param transportService The OpenSearch transport service
+     * @param discoveryExtension The extension node to which to send actions
+     */
+    public RestSendToExtensionAction(
+        RegisterRestActionsRequest restActionsRequest,
+        DiscoveryExtension discoveryExtension,
+        TransportService transportService
+    ) {
+        this.uriPrefix = "/_extensions/_" + restActionsRequest.getUniqueId();
+        List<Route> restActionsAsRoutes = new ArrayList<>();
+        for (String restAction : restActionsRequest.getRestActions()) {
+            RestRequest.Method method;
+            String uri;
+            try {
+                int delim = restAction.indexOf(' ');
+                method = RestRequest.Method.valueOf(restAction.substring(0, delim));
+                uri = uriPrefix + restAction.substring(delim).trim();
+            } catch (IndexOutOfBoundsException | IllegalArgumentException e) {
+                throw new IllegalArgumentException(restAction + " does not begin with a valid REST method");
+            }
+            logger.info("Registering: " + method + " " + uri);
+            restActionsAsRoutes.add(new Route(method, uri));
+        }
+        this.routes = unmodifiableList(restActionsAsRoutes);
+        this.discoveryExtension = discoveryExtension;
+        this.transportService = transportService;
+    }
+
+    @Override
+    public String getName() {
+        return "send_to_extension_action";
+    }
+
+    @Override
+    public List<Route> routes() {
+        return this.routes;
+    }
+
+    @Override
+    public RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
+        Method method = request.getHttpRequest().method();
+        String uri = request.getHttpRequest().uri();
+        if (uri.startsWith(uriPrefix)) {
+            uri = uri.substring(uriPrefix.length());
+        }
+        String message = "Forwarding the request " + method + " " + uri + " to " + discoveryExtension;
+        logger.info(message);
+        try {
+            // Notify user the request was acted on.
+            return channel -> channel.sendResponse(new BytesRestResponse(RestStatus.ACCEPTED, message));
```

Comment:
Still wrapping my head around this. 
When the response gets back is `return new RestExecuteOnExtensionResponse(in);` this returned to the consumer ?

Reply:
No, that's just logged internally.  This is a somewhat immediate response to the user that their request was accepted.  It sends the "message" parameter and a 202 code.

See the walk-through in [this comment](https://github.com/opensearch-project/opensearch-sdk-java/issues/64#issuecomment-1227494641):
```
(16:15:41) <0> [~] % curl localhost:9200/_extensions/_opensearch-sdk-1/api_1
Forwarding the request GET /api_1 to {sample-extension}{opensearch-sdk-1}{opensearch-sdk-1}{sdk_host}{127.0.0.1:4532}{dimr}%
```

This is compared to various 40x errors if it's a bad REST request.

In theory we could make the user wait (how long?) to see if the request was delivered to the extension and it responded (but do you respond on receipt or on finishing processing, etc.).

Reply:
I understand we dont want to let the user wait, I am trying to understand how does user get the response after the extension has processed it?

Reply:
That's a great question and is "step 9" per the issue diagram in [SDK #64](https://github.com/opensearch-project/opensearch-sdk-java/issues/64) which is outside the scope of this PR.  If it's ok to leave it like this for now, when I do a PR to address [SDK #102](https://github.com/opensearch-project/opensearch-sdk-java/issues/102), that will provide a "response" to the execute on extension action that we can give to the user.... basically moving the stuff in the finally block below this into a method and returning that response to the user.

Reply:
I've added commit dfad6d47edf30b246292f957e387b9b66bf5612f to address this.  (I actually have completed the work to implement [SDK #102](https://github.com/opensearch-project/opensearch-sdk-java/issues/102) and have tested this as part of it, that PR will come soon, but this part can now be included.   A lot of whitespace noise in that PR by unwrapping a try/finally, the real change is in the last 7 or 8 lines or so.